### PR TITLE
Assignment 06

### PR DESCRIPTION
**Pull request by as-ling**
Screenshot 2021-06-02 at 01 53 55
Screenshot 2021-06-02 at 01 55 00

Scalar field generated by making a idealised sun path (sunpath_mesh) around my object mesh. The field is made by finding, from each vertex in the object mesh, the closest points in my sun path, which would have the most 'sun' at that time. 

### Checklist

- [ ] Paste a screenshot of the solution whenever possible (`ctrl+v` or `cmd+v` directly on this textbox works)
- [ ] Add optional notes/remarks to be taken into account for evaluating the submission

### CODE REVIEW

**Comment by ioannaMitropoulou** [comment]
Hello, thanks for your submission!
This is an interesting application of the generation of contours from scalar field, this time not for 3d printing, but for an analysis of the model with relation to its relation with the 'sun'. The submission is complete and nicely documented. 

Regarding your question, I am not sure I understand what exactly you were trying to do, but I can describe how to get the closest normal (or any other 'closest' attribute) of the mesh from a point. 
This can be a challenge, because when you use the function `slicer_utils.get_closest_pt()` then you only get back the closest point itself, but no other information. What would be more useful instead, would be to have the **index** of the point, with which we can then access all the other information related to it. Here's two suggestions for how to get that.

* The first is, instead of the `slicer_utils.get_closest_pt()`  to use the compas function `compas.geometry.closest_point_in_cloud()` which returns the distance, the point, and its **index**. 
Here's an example. If we suppose that we have a point: `pt ` and a list of points: `vertex_coordinates`, then we can do the following:

```
closest_dist, closest_pt, closest_index = compas.geometry.closest_point_in_cloud(pt , vertex_coordinates)
closest_pt_index = vertex_coordinates[closest_index]
closest_vertex_key = mesh.index_key(closest_vertex_index )
closest_normal = mesh.vertex_normal(closest_vertex_key) # here we can get the normal, or any other information of the closest vertex
```

* The second option, is to search for the index of the point that the function `slicer_utils.get_closest_pt()` returns. There's a python function: `list.index(element)` for doing that. So in this context it would work in the following way:
```
closest_pt = slicer_utils.get_closest_pt(pt , vertex_coordinates)
closest_pt_index = vertex_coordinates.index(closest_pt)
closest_vertex_key = mesh.index_key(closest_pt_index)
closest_normal = mesh.vertex_normal(closest_vertex_key)
```
So, to sum up, it's a question of how to get the index of the closest point, instead of the closest point itself. Once we have the index, then we can access all its information, such as normal etc. I hope this answers the question! Let me know if something is unclear

Best,
Ioanna
